### PR TITLE
begin adding oauth docs

### PR DESCRIPTION
**Pull request by scottusrobus**

### CODE REVIEW

File: source/includes/_api.auth.md [hunk 1]
```diff
@@ -37,3 +39,83 @@ var vhxjs = new vhx('YOUR_API_KEY');
...
+> <span class="is-internal">Example Access Token Request</span>
+
+```plaintextinternal
+$ curl -X POST "https://api.vhx.tv/oauth/token" \
+  -d client_id=YOUR_CLIENT_ID \
+  -d client_secret=YOUR_CLIENT_SECRET \
+  -d grant_type=PASSWORD \
+  -d username=YOUR_VHX_EMAIL \
+  -d password=YOUR_VHX_PASSWORD
+```
```
Comment (ksheurs) [inline source/includes/_api.auth.md:90]:
```plaintextinternal
$ curl -X POST "https://api.vhx.tv/oauth/token" \
   -d client_id=YOUR_CLIENT_ID \
   -d client_secret=YOUR_CLIENT_SECRET \
   -d grant_type=client_credentials
```

File: source/includes/_api.auth.md [hunk 2]
```diff
@@ -37,3 +39,83 @@ var vhxjs = new vhx('YOUR_API_KEY');
...
+
+> <span class="is-internal">Example Response</span>
+
+```json
+{
+  "access_token": "YOUR_NEW_ACCESS_TOKEN",
+  "token_type": "bearer",
+  "expires_in": 7200,
+  "refresh_token": "YOUR_NEW_REFRESH_TOKEN",
+  "scope": "public"
```
Comment (ksheurs) [inline source/includes/_api.auth.md:136]: use `"read"` for all of these

**Comment by scottusrobus** [comment]
@ksheurs 
this is ready for your review! I added the remaining OAuth documentation and examples.

I've deployed it to make it easier to take look (behind the internal flag).
http://dev.vhx.tv/docs/api/?internal=1#authentication-oauth

File: source/includes/_api.auth.md [hunk 3]
```diff
@@ -37,3 +39,114 @@ var vhxjs = new vhx('YOUR_API_KEY');
...
+
+  <h4><strong>2. Poll for user activation</strong></h4>
+  <p>Upon displaying the code to the customer, you should begin polling <code>/oauth/codes/{code}</code> for the code at a ~2.5 second interval with a max of 500 retries.</p>
+  <p>This will return a <code>404</code> until the customer correctly pairs it with their account. At that point a <code>200</code> will be returned with an <code>access_token</code>, <code>refresh_token</code>, and <code>expiry</code> information. This information should be saved in your device's registry and is what deems the device as being connected/linked to a logged in customer.</p>
+  <p>You can than make requests on behalf of the customer using the Authorization Bearer header.</p>
+
+  <h4><strong>3. Refresh Tokens</strong></h4>
+  <p>Access tokens for a user expire after 2 hours. To continue making API requests you can refresh your access token by requesting a new one (from the <code>/oauth/token</code> endpoint) using the provided <code>refresh_token</code> in the previous response.</p>
+</section>
```
Comment (ksheurs) [inline source/includes/_api.auth.md:153]: there are lot's of references to "user" here. let's maybe try and stick with "customer" ?

File: source/includes/_api.auth.md [hunk 4]
```diff
@@ -37,3 +39,114 @@ var vhxjs = new vhx('YOUR_API_KEY');
...
+     -d client_id={client_id}
+     -d client_secret={client_secret}
+```
+
+> <span class="is-internal">Example Response</span>
+
+```jsoninternal
+{
+  "code": "YOUR_PAIRING_CODE"
+}
```
Comment (ksheurs) [inline source/includes/_api.auth.md:98]: this also returns an "expires_in". example:

```json
{
  "code": "DDCD",
  "expires_in": 1800
}
```

File: source/includes/_api.auth.md (hunk 2 above)
Comment (ksheurs) [reply source/includes/_api.auth.md:136]: missing "read"

**Comment by ksheurs** [comment]
Left some feedback, @scottusrobus ! Question - that also came up within some iOS player SDK talk. I forget, but does our client libraries support both access token and api key? or just api key right now? if the latter, what are your thoughts on the matter?

**Comment by scottusrobus** [comment]
thanks for reviewing!
Currently only supports API Keys. In general I think the oauth flows probably aren't as useful on web clients, as using API keys is simpler and our preferred method.

But I do see it being more useful for iOS or device apps, particularly where pairing codes might be used.

**Comment by ksheurs** [comment]
@scottusrobus yea it's tricky where refresh logic should live. think it's probably a concern that lives outside of the client libraries. and it's up to you to pass in a valid / non-expired access token.

**Comment by scottusrobus** [comment]
@ksheurs yeah, it is tricky. agreed i think that makes sense.

**Pull request merged**